superset/stackable/opa-authorizer/tests/opa_manager_test.py

```diff
@@ -0,0 +1,247 @@
+    """
+    opa_roles = []
+
+    with app.app_context():
+        mocker.patch(
+            "flask_appbuilder.security.sqla.manager.SecurityManager.update_user",
+            return_value=True,
+        )
+        mocker.patch(
+            "opa_authorizer.opa_manager.OpaSupersetSecurityManager.get_opa_user_roles",
+            return_value=opa_roles,
+        )
+        mocker.patch(
+            "opa_authorizer.opa_manager.OpaSupersetSecurityManager.resolve_role",
+            wraps=mock_resolve_role,
+        )
+
+        assert set(
+            map(lambda r: r.name, opa_security_manager.get_user_roles(user))
+        ) == {"Public"}
+
+
+def test_get_opa_roles(
+    mocker: MockFixture,
+    app: Flask,
+    opa_security_manager: OpaSupersetSecurityManager,
+) -> None:
+    """
+    Test that roles are correctly extracted from the OPA response.
+    """
+    response = requests.Response()
+    response.status_code = 200
+    response._content = '{"result": ["Test1", "Test2", "Test3"]}'.encode()
+
+    with app.app_context():
+        mocker.patch(
+            "opa_authorizer.opa_manager.OpaSupersetSecurityManager.call_opa",
+            return_value=response,
+        )
+
+        assert opa_security_manager.get_opa_user_roles("User1") == [
+            "Test1",
+            "Test2",
+            "Test3",
+        ]
+
+
+def test_get_opa_roles_no_result(
+    mocker: MockFixture,
+    app: Flask,
+    opa_security_manager: OpaSupersetSecurityManager,
+) -> None:
+    """
+    Test that no roles are returned if the OPA response doesn't contain a query result.
+    """
+    response = requests.Response()
+    response.status_code = 200
+    response._content = '{"error": "error occurred"}'.encode()
+
+    with app.app_context():
+        mocker.patch(
+            "opa_authorizer.opa_manager.OpaSupersetSecurityManager.call_opa",
+            return_value=response,
+        )
+
+        assert opa_security_manager.get_opa_user_roles("User1") == []
+
+
+def test_get_opa_roles_not_a_list(
+    mocker: MockFixture,
+    app: Flask,
+    opa_security_manager: OpaSupersetSecurityManager,
+) -> None:
+    """
+    Test that no roles are returned if the query result doesn't contain a list.
+    """
+    response = requests.Response()
+    response.status_code = 200
+    response._content = '{"result": "some string"}'.encode()
+
+    with app.app_context():
+        mocker.patch(
+            "opa_authorizer.opa_manager.OpaSupersetSecurityManager.call_opa",
+            return_value=response,
+        )
+
+        assert opa_security_manager.get_opa_user_roles("User1") == []
+
+
+def test_get_opa_roles_wrong_statuscode(
+    mocker: MockFixture,
+    app: Flask,
+    opa_security_manager: OpaSupersetSecurityManager,
+) -> None:
+    """
+    Test that no roles are returned if the query result doesn't contain a list.
+    """
+    response = requests.Response()
+    response.status_code = 200
```

Comment:
The status code 200 is not "wrong" as stated in the test function name.

Please add also a test case where an exception is thrown by the requests. A real-world example would be an SSLError.

Reply:
The function name `test_get_opa_roles_wrong_statuscode` states that a wrong status code is returned, i.e. not a 200 OK. On the other hand, the comment does not mention the status code but states that the result does not contain a list. Please make this consistent, e.g by changing the comment and setting the status code to 500.